src/cljs/athens/components/default_components.cljs

```diff
@@ -0,0 +1,62 @@
+(ns athens.components.default-components
+  (:require
+    [athens.db :as db]
+    [athens.util :refer [now-ts]]
+    [re-frame.core :refer [dispatch]]))
+
+;; Note to contributors: After you define the component, you should add it in the exported components vector at bottom.
+
+;; ---- Helper functions for default components ----
+(defn todo-on-click
+  [uid from-str to-str]
+  (let [current-block-content (:block/string (db/get-block [:block/uid uid]))]
+    (dispatch [:transact [{:block/uid    uid
+                           :block/string (clojure.string/replace
+                                           current-block-content
+                                           from-str
+                                           to-str)
+                           :edit/time    (now-ts)}]])))
+
+
+(defn find-weblink
+  [content]
+  (re-find #"http.*" content))
+
+
+;; ---- Todo component declaration ----
+(def component-todo
+  {:match #"\[\[TODO\]\]"
+   :render (fn [_ uid]
+             [:input {:type     "checkbox"
+                      :on-click #(todo-on-click uid #"\{\{\[\[TODO\]\]\}\}" "{{[[DONE]]}}")}])})
+
+
+(def component-done
+  {:match #"\[\[DONE\]\]"
+   :render (fn [_ uid]
+             [:input {:type     "checkbox"
+                      :checked  "true"
+                      :on-click #(todo-on-click uid #"\{\{\[\[DONE\]\]\}\}" "{{[[TODO]]}}")}])})
+
+
+;; ---- Website embed component declaration ----
+(def component-youtube-embed
+  {:match  #"\[\[youtube\]\]\:.*"
+   :render (fn [content _]
+             [:iframe {:width       640
+                       :height      360
+                       :src         (str "https://www.youtube.com/embed/" (get (re-find #".*v=([a-zA-Z0-9_\-]+)" content) 1))
+                       :allow       "accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"}])})
+
+
+(def component-generic-embed
+  {:match  #"iframe\:.*"
```

Comment:
Is it on purpose that youtube is `{{[[youtube: ]]}}` but iframe is `{{iframe: }}`? Should we be consistent?

Reply:
This is kind of a legacy problem from Roam - their component is exactly like how I implemented. I do believe we might need a revamp on the syntax & data structure of Roam for consistency, but in the current method's defense, YouTube is a more specific website, which makes having backlinks to YouTube make more sense than just backlinking every iframe embeds.